### DISCUSSION TITLE
Option to turn off dynamic positioning

### DISCUSSION THREAD

**Discussion by asacarter** [taufik-nurrohman/color-picker#63] [Ideas]
I'm attaching the the picker to the elements parent and positioning it relative to that.

``` .js
document.querySelectorAll('.colorpicker').forEach(source => {
    let picker = new CP(source, { parent: source.parentElement });
    insertColorCode(picker);
});
```

I override the inline style with `.color-picker{position:absolute;top:20px!important;left:0!important;`

This works fine but it would nice to have the option to turn off the dynamic inline style positioning.

**Comment by taufik-nurrohman** [comment]
I would probably add a `fit` option in the future to disable it. Something like this:

``` .js
const picker = new CP(source, {
    fit: [0, 0], // Force `left` and `top` position to `0`
    parent: source.parentNode
});
```

Or by changing the `fit` hook logic to follow the `blur` and `focus` hook principle. So, instead of firing a function on every `fit` event, it will simply remove the default `fit` behaviour. So users could make their own `fit` logic:

~~~ .js
picker.on('fit', () => {
    picker.self.style.left = 0;
    picker.self.style.top = 0;
});
~~~

**Comment by asacarter** [comment]
Sounds good to me :)

Another good idea would be to use class names rather than `:sv` and `:h` for example. I had to modify the source code so that I could use `:before` and `:after` on these elements.

**Comment by taufik-nurrohman** [comment]
Use pseudo class and element this way:

~~~ .css
.color-picker\:h::before {}
.color-picker\:h:hover {}
~~~
